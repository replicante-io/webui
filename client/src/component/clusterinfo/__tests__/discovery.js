'use strict';

import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Discovery from '../discovery';


const mockStore = configureStore([]);
const MATCH = {
  params: {name: 'test'}
};


describe('ClusterInfo', () => {
  describe('discovery', () => {
    let store;

    beforeEach(() => {
      store = mockStore({
        clusterinfo: {
          discovery: {}
        }
      });
    });

    test('renders no data', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <Discovery match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders with data', () => {
      store = mockStore({ clusterinfo: { discovery: {
        'test': {
          cluster: 'test',
          nodes: [
            'http://node1',
            'http://node2'
          ]
        }
      } } });
      const tree = renderer.create(
        <Provider store={store}>
          <Discovery match={MATCH} />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
