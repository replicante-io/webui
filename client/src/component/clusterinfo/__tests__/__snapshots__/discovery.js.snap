// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterInfo discovery renders no data 1`] = `
<div
  className="text-center"
>
  <p
    className="large-icon nodata"
  >
    <span
      className="fa-layers fa-fw"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-database "
        data-icon="database"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 80v48c0 44.2-100.3 80-224 80S0 172.2 0 128V80C0 35.8 100.3 0 224 0S448 35.8 448 80zM393.2 214.7c20.8-7.4 39.9-16.9 54.8-28.6V288c0 44.2-100.3 80-224 80S0 332.2 0 288V186.1c14.9 11.8 34 21.2 54.8 28.6C99.7 230.7 159.5 240 224 240s124.3-9.3 169.2-25.3zM0 346.1c14.9 11.8 34 21.2 54.8 28.6C99.7 390.7 159.5 400 224 400s124.3-9.3 169.2-25.3c20.8-7.4 39.9-16.9 54.8-28.6V432c0 44.2-100.3 80-224 80S0 476.2 0 432V346.1z"
          fill="currentColor"
          style={{}}
        />
      </svg>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-question overlay"
        data-icon="question"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={{}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 96c-17.7 0-32 14.3-32 32s-14.3 32-32 32s-32-14.3-32-32C0 75 43 32 96 32h97c70.1 0 127 56.9 127 127c0 52.4-32.2 99.4-81 118.4l-63 24.5 0 18.1c0 17.7-14.3 32-32 32s-32-14.3-32-32V301.9c0-26.4 16.2-50.1 40.8-59.6l63-24.5C240 208.3 256 185 256 159c0-34.8-28.2-63-63-63H96zm48 384c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
          fill="currentColor"
          style={{}}
        />
      </svg>
    </span>
  </p>
  No information available. 
  <br />
  If nothing shows up the cluster may not exist.
</div>
`;

exports[`ClusterInfo discovery renders with data 1`] = `
<div
  className="codeblock"
>
  <div
    className="react-json-view"
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "cursor": "default",
        "fontFamily": "monospace",
        "position": "relative",
      }
    }
  >
    <div
      className="pretty-json-container object-container"
    >
      <div
        className="object-content"
      >
        <div
          className="object-key-val"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span>
            <span
              onClick={[Function]}
              style={
                {
                  "cursor": "pointer",
                  "display": "inline-block",
                }
              }
            >
              <div
                className="icon-container"
                style={
                  {
                    "display": "inline-block",
                    "width": "17px",
                  }
                }
              >
                <span
                  className="expanded-icon"
                >
                  <svg
                    fill="currentColor"
                    style={
                      {
                        "color": "#586e75",
                        "height": "1em",
                        "paddingLeft": "2px",
                        "verticalAlign": "top",
                        "width": "1em",
                      }
                    }
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 5l6 6 6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span />
              <span
                style={
                  {
                    "color": "#002b36",
                    "cursor": "pointer",
                    "display": "inline-block",
                    "fontWeight": "bold",
                  }
                }
              >
                {
              </span>
            </span>
            <div
              className="object-meta-data"
              onClick={[Function]}
              style={
                {
                  "display": "inline-block",
                  "padding": "0px 0px 0px 10px",
                }
              }
            >
              <span
                className="object-size"
                style={
                  {
                    "borderRadius": "3px",
                    "color": "rgba(0, 0, 0, 0.3)",
                    "cursor": "default",
                    "fontStyle": "italic",
                    "margin": "0px 6px 0px 0px",
                  }
                }
              >
                3
                 item
                s
              </span>
              <span
                className="copy-to-clipboard-container"
                style={
                  {
                    "display": "none",
                    "verticalAlign": "top",
                  }
                }
                title="Copy to clipboard"
              >
                <span
                  onClick={[Function]}
                  style={
                    {
                      "cursor": "pointer",
                      "display": "inline",
                    }
                  }
                >
                  <span
                    className="copy-icon"
                  >
                    <svg
                      fill="currentColor"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        {
                          "color": "#268bd2",
                          "fontSize": "15px",
                          "height": "1em",
                          "marginRight": "3px",
                          "verticalAlign": "top",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 40 40"
                    >
                      <g>
                        <path
                          d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                        />
                      </g>
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </span>
          <div
            className="pushed-content object-container"
          >
            <div
              className="object-content"
              style={
                {
                  "marginLeft": "6px",
                }
              }
            >
              <div
                className="variable-row"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    ":hover": {
                      "borderLeft": "2px solid rgb(235, 235, 235)",
                      "paddingLeft": "19px",
                    },
                    "borderLeft": "1px solid rgb(235, 235, 235)",
                    "paddingBottom": "3px",
                    "paddingLeft": 20,
                    "paddingRight": "5px",
                    "paddingTop": "3px",
                  }
                }
              >
                <span>
                  <span
                    className="object-key"
                    style={
                      {
                        ":hover": {
                          "opacity": "1",
                        },
                        "color": "#002b36",
                        "display": "inline-block",
                        "fontStyle": "none",
                        "letterSpacing": "0.5px",
                        "opacity": "0.85",
                        "verticalAlign": "top",
                      }
                    }
                  >
                    <span
                      style={
                        {
                          "verticalAlign": "top",
                        }
                      }
                    >
                      "
                    </span>
                    <span
                      style={
                        {
                          "display": "inline-block",
                        }
                      }
                    >
                      cluster_id
                    </span>
                    <span
                      style={
                        {
                          "verticalAlign": "top",
                        }
                      }
                    >
                      "
                    </span>
                  </span>
                  <span
                    style={
                      {
                        "color": "#002b36",
                        "display": "inline-block",
                        "margin": "0px 5px",
                        "verticalAlign": "top",
                      }
                    }
                  >
                    :
                  </span>
                </span>
                <div
                  className="variable-value"
                  onClick={null}
                  style={
                    {
                      "cursor": "default",
                      "display": "inline-block",
                      "paddingRight": "6px",
                      "position": "relative",
                    }
                  }
                >
                  <div
                    style={
                      {
                        "color": "#cb4b16",
                        "display": "inline-block",
                      }
                    }
                  >
                    <span
                      className="data-type-label"
                      style={
                        {
                          "fontSize": "11px",
                          "marginRight": "4px",
                          "opacity": "0.8",
                        }
                      }
                    >
                      string
                    </span>
                    <span
                      className="string-value"
                      onClick={[Function]}
                      style={
                        {
                          "cursor": "default",
                        }
                      }
                    >
                      "
                      test
                      "
                    </span>
                  </div>
                </div>
                <span
                  className="copy-to-clipboard-container"
                  style={
                    {
                      "display": "none",
                      "verticalAlign": "top",
                    }
                  }
                  title="Copy to clipboard"
                >
                  <span
                    onClick={[Function]}
                    style={
                      {
                        "cursor": "pointer",
                        "display": "inline",
                      }
                    }
                  >
                    <span
                      className="copy-icon"
                    >
                      <svg
                        fill="currentColor"
                        preserveAspectRatio="xMidYMid meet"
                        style={
                          {
                            "color": "#268bd2",
                            "fontSize": "15px",
                            "height": "1em",
                            "marginRight": "3px",
                            "verticalAlign": "top",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 40 40"
                      >
                        <g>
                          <path
                            d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                          />
                        </g>
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
              <div
                className="variable-row"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    ":hover": {
                      "borderLeft": "2px solid rgb(235, 235, 235)",
                      "paddingLeft": "19px",
                    },
                    "borderLeft": "1px solid rgb(235, 235, 235)",
                    "paddingBottom": "3px",
                    "paddingLeft": 20,
                    "paddingRight": "5px",
                    "paddingTop": "3px",
                  }
                }
              >
                <span>
                  <span
                    className="object-key"
                    style={
                      {
                        ":hover": {
                          "opacity": "1",
                        },
                        "color": "#002b36",
                        "display": "inline-block",
                        "fontStyle": "none",
                        "letterSpacing": "0.5px",
                        "opacity": "0.85",
                        "verticalAlign": "top",
                      }
                    }
                  >
                    <span
                      style={
                        {
                          "verticalAlign": "top",
                        }
                      }
                    >
                      "
                    </span>
                    <span
                      style={
                        {
                          "display": "inline-block",
                        }
                      }
                    >
                      display_name
                    </span>
                    <span
                      style={
                        {
                          "verticalAlign": "top",
                        }
                      }
                    >
                      "
                    </span>
                  </span>
                  <span
                    style={
                      {
                        "color": "#002b36",
                        "display": "inline-block",
                        "margin": "0px 5px",
                        "verticalAlign": "top",
                      }
                    }
                  >
                    :
                  </span>
                </span>
                <div
                  className="variable-value"
                  onClick={null}
                  style={
                    {
                      "cursor": "default",
                      "display": "inline-block",
                      "paddingRight": "6px",
                      "position": "relative",
                    }
                  }
                >
                  <div
                    style={
                      {
                        "backgroundColor": "rgb(235, 235, 235)",
                        "borderRadius": "3px",
                        "color": "#dc322f",
                        "display": "inline-block",
                        "fontSize": "11px",
                        "fontWeight": "bold",
                        "padding": "1px 2px",
                      }
                    }
                  >
                    NULL
                  </div>
                </div>
                <span
                  className="copy-to-clipboard-container"
                  style={
                    {
                      "display": "none",
                      "verticalAlign": "top",
                    }
                  }
                  title="Copy to clipboard"
                >
                  <span
                    onClick={[Function]}
                    style={
                      {
                        "cursor": "pointer",
                        "display": "inline",
                      }
                    }
                  >
                    <span
                      className="copy-icon"
                    >
                      <svg
                        fill="currentColor"
                        preserveAspectRatio="xMidYMid meet"
                        style={
                          {
                            "color": "#268bd2",
                            "fontSize": "15px",
                            "height": "1em",
                            "marginRight": "3px",
                            "verticalAlign": "top",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 40 40"
                      >
                        <g>
                          <path
                            d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                          />
                        </g>
                      </svg>
                    </span>
                  </span>
                </span>
              </div>
              <div
                className="object-key-val"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    ":hover": {
                      "borderLeft": "2px solid rgb(235, 235, 235)",
                      "paddingLeft": "19px",
                    },
                    "borderLeft": "1px solid rgb(235, 235, 235)",
                    "paddingBottom": "3px",
                    "paddingLeft": 20,
                    "paddingRight": "5px",
                    "paddingTop": "3px",
                  }
                }
              >
                <span>
                  <span
                    onClick={[Function]}
                    style={
                      {
                        "cursor": "pointer",
                        "display": "inline-block",
                      }
                    }
                  >
                    <div
                      className="icon-container"
                      style={
                        {
                          "display": "inline-block",
                          "width": "17px",
                        }
                      }
                    >
                      <span
                        className="expanded-icon"
                      >
                        <svg
                          fill="currentColor"
                          style={
                            {
                              "color": "#586e75",
                              "height": "1em",
                              "paddingLeft": "2px",
                              "verticalAlign": "top",
                              "width": "1em",
                            }
                          }
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 5l6 6 6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span
                      style={
                        {
                          ":hover": {
                            "opacity": "1",
                          },
                          "color": "#002b36",
                          "display": "inline-block",
                          "fontStyle": "none",
                          "letterSpacing": "0.5px",
                          "opacity": "0.85",
                          "verticalAlign": "top",
                        }
                      }
                    >
                      <span
                        className="object-key"
                      >
                        <span
                          style={
                            {
                              "verticalAlign": "top",
                            }
                          }
                        >
                          "
                        </span>
                        <span>
                          nodes
                        </span>
                        <span
                          style={
                            {
                              "verticalAlign": "top",
                            }
                          }
                        >
                          "
                        </span>
                      </span>
                      <span
                        style={
                          {
                            "color": "#002b36",
                            "display": "inline-block",
                            "margin": "0px 5px",
                            "verticalAlign": "top",
                          }
                        }
                      >
                        :
                      </span>
                    </span>
                    <span
                      style={
                        {
                          "color": "#002b36",
                          "cursor": "pointer",
                          "display": "inline-block",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      [
                    </span>
                  </span>
                  <div
                    className="object-meta-data"
                    onClick={[Function]}
                    style={
                      {
                        "display": "inline-block",
                        "padding": "0px 0px 0px 10px",
                      }
                    }
                  >
                    <span
                      className="object-size"
                      style={
                        {
                          "borderRadius": "3px",
                          "color": "rgba(0, 0, 0, 0.3)",
                          "cursor": "default",
                          "fontStyle": "italic",
                          "margin": "0px 6px 0px 0px",
                        }
                      }
                    >
                      2
                       item
                      s
                    </span>
                    <span
                      className="copy-to-clipboard-container"
                      style={
                        {
                          "display": "none",
                          "verticalAlign": "top",
                        }
                      }
                      title="Copy to clipboard"
                    >
                      <span
                        onClick={[Function]}
                        style={
                          {
                            "cursor": "pointer",
                            "display": "inline",
                          }
                        }
                      >
                        <span
                          className="copy-icon"
                        >
                          <svg
                            fill="currentColor"
                            preserveAspectRatio="xMidYMid meet"
                            style={
                              {
                                "color": "#268bd2",
                                "fontSize": "15px",
                                "height": "1em",
                                "marginRight": "3px",
                                "verticalAlign": "top",
                                "width": "1em",
                              }
                            }
                            viewBox="0 0 40 40"
                          >
                            <g>
                              <path
                                d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                              />
                            </g>
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </span>
                <div
                  className="pushed-content object-container"
                >
                  <div
                    className="object-content"
                    style={
                      {
                        "marginLeft": "6px",
                      }
                    }
                  >
                    <div
                      className="variable-row"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        {
                          ":hover": {
                            "borderLeft": "2px solid rgb(235, 235, 235)",
                            "paddingLeft": "19px",
                          },
                          "borderLeft": "1px solid rgb(235, 235, 235)",
                          "paddingBottom": "3px",
                          "paddingLeft": 20,
                          "paddingRight": "5px",
                          "paddingTop": "3px",
                        }
                      }
                    >
                      <span
                        style={
                          {
                            ":hover": {
                              "opacity": "1",
                            },
                            "color": "#6c71c4",
                            "display": "inline-block",
                            "fontStyle": "none",
                            "letterSpacing": "0.5px",
                            "opacity": "0.85",
                            "verticalAlign": "top",
                          }
                        }
                      >
                        0
                        <div
                          style={
                            {
                              "color": "#002b36",
                              "display": "inline-block",
                              "margin": "0px 5px",
                              "verticalAlign": "top",
                            }
                          }
                        >
                          :
                        </div>
                      </span>
                      <div
                        className="variable-value"
                        onClick={null}
                        style={
                          {
                            "cursor": "default",
                            "display": "inline-block",
                            "paddingRight": "6px",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "color": "#cb4b16",
                              "display": "inline-block",
                            }
                          }
                        >
                          <span
                            className="data-type-label"
                            style={
                              {
                                "fontSize": "11px",
                                "marginRight": "4px",
                                "opacity": "0.8",
                              }
                            }
                          >
                            string
                          </span>
                          <span
                            className="string-value"
                            onClick={[Function]}
                            style={
                              {
                                "cursor": "default",
                              }
                            }
                          >
                            "
                            http://node1
                            "
                          </span>
                        </div>
                      </div>
                      <span
                        className="copy-to-clipboard-container"
                        style={
                          {
                            "display": "none",
                            "verticalAlign": "top",
                          }
                        }
                        title="Copy to clipboard"
                      >
                        <span
                          onClick={[Function]}
                          style={
                            {
                              "cursor": "pointer",
                              "display": "inline",
                            }
                          }
                        >
                          <span
                            className="copy-icon"
                          >
                            <svg
                              fill="currentColor"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                {
                                  "color": "#268bd2",
                                  "fontSize": "15px",
                                  "height": "1em",
                                  "marginRight": "3px",
                                  "verticalAlign": "top",
                                  "width": "1em",
                                }
                              }
                              viewBox="0 0 40 40"
                            >
                              <g>
                                <path
                                  d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                />
                              </g>
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                    <div
                      className="variable-row"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        {
                          ":hover": {
                            "borderLeft": "2px solid rgb(235, 235, 235)",
                            "paddingLeft": "19px",
                          },
                          "borderLeft": "1px solid rgb(235, 235, 235)",
                          "paddingBottom": "3px",
                          "paddingLeft": 20,
                          "paddingRight": "5px",
                          "paddingTop": "3px",
                        }
                      }
                    >
                      <span
                        style={
                          {
                            ":hover": {
                              "opacity": "1",
                            },
                            "color": "#6c71c4",
                            "display": "inline-block",
                            "fontStyle": "none",
                            "letterSpacing": "0.5px",
                            "opacity": "0.85",
                            "verticalAlign": "top",
                          }
                        }
                      >
                        1
                        <div
                          style={
                            {
                              "color": "#002b36",
                              "display": "inline-block",
                              "margin": "0px 5px",
                              "verticalAlign": "top",
                            }
                          }
                        >
                          :
                        </div>
                      </span>
                      <div
                        className="variable-value"
                        onClick={null}
                        style={
                          {
                            "cursor": "default",
                            "display": "inline-block",
                            "paddingRight": "6px",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          style={
                            {
                              "color": "#cb4b16",
                              "display": "inline-block",
                            }
                          }
                        >
                          <span
                            className="data-type-label"
                            style={
                              {
                                "fontSize": "11px",
                                "marginRight": "4px",
                                "opacity": "0.8",
                              }
                            }
                          >
                            string
                          </span>
                          <span
                            className="string-value"
                            onClick={[Function]}
                            style={
                              {
                                "cursor": "default",
                              }
                            }
                          >
                            "
                            http://node2
                            "
                          </span>
                        </div>
                      </div>
                      <span
                        className="copy-to-clipboard-container"
                        style={
                          {
                            "display": "none",
                            "verticalAlign": "top",
                          }
                        }
                        title="Copy to clipboard"
                      >
                        <span
                          onClick={[Function]}
                          style={
                            {
                              "cursor": "pointer",
                              "display": "inline",
                            }
                          }
                        >
                          <span
                            className="copy-icon"
                          >
                            <svg
                              fill="currentColor"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                {
                                  "color": "#268bd2",
                                  "fontSize": "15px",
                                  "height": "1em",
                                  "marginRight": "3px",
                                  "verticalAlign": "top",
                                  "width": "1em",
                                }
                              }
                              viewBox="0 0 40 40"
                            >
                              <g>
                                <path
                                  d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                                />
                              </g>
                            </svg>
                          </span>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  className="brace-row"
                >
                  <span
                    style={
                      {
                        "color": "#002b36",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontWeight": "bold",
                        "paddingLeft": "3px",
                      }
                    }
                  >
                    ]
                  </span>
                </span>
              </div>
            </div>
          </div>
          <span
            className="brace-row"
          >
            <span
              style={
                {
                  "color": "#002b36",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontWeight": "bold",
                  "paddingLeft": "3px",
                }
              }
            >
              }
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
